analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RegularExpressions/RegexShouldNotContainMultipleSpaces.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+class Compliant
+{
+    void Ctor()
+    {
+        var defaultOrder = new Regex("single space"); // Compliant
+
+        var namedArgs = new Regex(
+            pattern: "single space");
+
+        var noRegex = new NoRegex("single space"); // Compliant
+
+        var singleOption = new Regex("ignore  pattern  white space", RegexOptions.IgnorePatternWhitespace); // Compliant
+        var mixedOptions = new Regex("ignore  pattern  white space", RegexOptions.IgnorePatternWhitespace | RegexOptions.Compiled); // Compliant
+    }
+
+    void Static()
+    {
+        var isMatch = Regex.IsMatch("some input", "single space"); // Compliant
+        var noRegex = NoRegex.IsMatch("some input", "multiple  white  spaces"); // Compliant
+    }
+
+    void Unknown(string unknown)
+    {
+        var regex = new NoRegex(unknown + "multiple  white  spaces"); // Compliant
+    }
+
+    bool ConcatanationMultiline(string input)
+    {
+        return Regex.IsMatch(input, "single space"
+            + "|b"
+            + "|c"
+            + "|d]"); // Compliant
+    }
+
+    bool ConcatanationSingleIne(string input)
+    {
+        return Regex.IsMatch(input, "a" + "|b" + "|c" + "|single white space"); // Compliant
+    }
+
+    [RegularExpression("single space")] // Compliant
+    public string Attribute { get; set; }
+
+    bool WhiteSpaceVariations(string input)
+    {
+        return Regex.IsMatch(input, " with multple single spaces ")
+            || Regex.IsMatch(input, "without_spaces")
+            || Regex.IsMatch(input, "with\ttab")
+            || Regex.IsMatch(input, "")
+            || Regex.IsMatch(input, "ignore  pattern  white space", RegexOptions.IgnorePatternWhitespace);
+    }
+}
+
+class Noncompliant
+{
+    private const string Prefix = ".*";
+
+    void Ctor()
+    {
+        var patternOnly = new Regex("multiple  white  spaces"); // Noncompliant {{Regular expressions should not contain multiple spaces.}}
```

Comment:
Let's make one of them way longer, to be sure we don't report twice
```suggestion
        var patternOnly = new Regex("multiple  white          spaces"); // Noncompliant {{Regular expressions should not contain multiple spaces.}}
```